src/index.js

```diff
@@ -1,27 +1,61 @@
 /**
  * @flow
  */
+import * as emails from './store/emails.json';
 
 // example of generating HTML with js
+function renderNewMessage(sender, timestamp, subject, snippet) {
+  const sidebar = document.querySelector('.email-list');
+  const newMessage = document.createElement('li');
+  newMessage.innerHTML = `<button class="email-item" type="button">
+                            <div class="sender-details">
+                              <p>${sender}</p>
+                              <span>${timestamp}</span>
+                            </div>
+                            <p class="email-subject">${subject}</p>
+                            <p>${snippet}</p>
+                          </button>`;
+  sidebar.appendChild(newMessage);
+}
+
+
+function formatTimestamp(datestring) {
+  const d = new Date(datestring);
+  const month = d.getMonth() + 1;
+  const day = d.getDay();
+  const year = String(d.getFullYear()).substring(2);
+  const timestamp = `${month}/${day}/${year}`;
+  return timestamp;
+}
+
+function formatSubject (subject) {
+  return subject.length > 33 ? subject.substring(0, 33) + '...' : subject;
+}
+
+function formatSnippet (snippet) {
+  return snippet.length > 77 ? snippet.substring(0, 77) + '...' : snippet;
+}
+
 function renderSidebar() {
   const sidebarContents = `
     <h2 class="email-header">Inbox</h2>
-    <ul class="email-list">
-      <li>
-        <button class="email-item" type="button">
-          <div class="sender-details">
-            <p>Email sender</p>
-            <span>Timestamp</span>
-          </div>
-          <p class="email-subject">Email1 subject</p>
-          <p>Email snippet</p>
-        </button>
-      </li>
-    </ul>
+    <ul class="email-list"></ul>
   `;
 
   const container = document.querySelector('.email-list-container');
   if (container != null) container.innerHTML = sidebarContents;
+
+  const newMessageIDs = emails.mailboxes.INBOX.threadIds;
+  const allMessages = emails.messages;
+
+  for (const m of newMessageIDs) {
+    const payloadHeaders = allMessages[m].payload.headers;
+    const snippet = formatSnippet(allMessages[m].snippet);
+    const timestamp = formatTimestamp(payloadHeaders[0].value);
+    const sender = payloadHeaders[1].value.match(/([A-Z])\w+/g).join(' ');
+    const subject = formatSubject(payloadHeaders[3].value);
```

Comment:
I would use a func more similar to Cameron's sample fn to grab the subject header of a particular email because relying on the position of this in the payloadHeaders falls over if any pieces of data are missing from the email :)